core/errors.py

```diff
@@ -18,3 +18,15 @@ def validation_error_handler(error: ValidationError):
         "status": 400,
         "title": "Bad Request",
     }, 400
+
+
+class UnimplementedUCException(Exception):
+    """Returned when an uncaught exception is raised"""
```

Comment:
I don't think the docstring is quite right. Something like this is more accurate IMO:
```
"""Raised when a validation error occurs that is not supported as a user-centred error message."""
```